compiler/verification/dune

```diff
@@ -1,7 +1,17 @@
 (library
  (name verification)
  (public_name catala.verification)
- (libraries bindlib utils dcalc runtime z3 calendar))
+ (libraries
+  bindlib
+  utils
+  dcalc
+  runtime
+  calendar
+  (select
+   z3backend.ml
+   from
+   (z3 -> z3backend.real.ml)
```

Comment:
Can you here make the case disjunction based on the fact that the user has passed a particular compilation flag or profile?